### PR TITLE
Add the jailed hero details dialog in Editor

### PR DESCRIPTION
**Pull request by Districh-ru**
Relates to #6845
This dialog is based on Hero Details dialog and allows to set the jailed hero race.
hero

### CODE REVIEW

File: src/fheroes2/maps/map_object_info.h [hunk 1]
```diff
@@ -158,6 +158,8 @@ namespace Maps
 
     const std::vector<ObjectInfo> & getObjectsByGroup( const ObjectGroup group );
 
+    const Maps::ObjectInfo & getObjectInfo( const Maps::ObjectGroup group, const int32_t objectType );
```
Comment (ihhub) [inline src/fheroes2/maps/map_object_info.h:161]: **objectType** should be **objectIndex**.

Comment (ihhub) [inline src/fheroes2/maps/map_object_info.h:161]: Please remove `Maps` enumeration usage in this line as this is already under the same enumeration.

File: src/fheroes2/maps/map_format_info.h [hunk 2]
```diff
@@ -133,17 +134,20 @@ namespace Maps::Map_Format
         // The amount of magic points (mana). Negative value means it is not set.
         int16_t magicPoints{ -1 };
 
+        // Jailed hero race, used only for jailed heroes. Race for normal heroes is stored in the ObjectInfo metadata.
+        uint8_t jailedHeroRace{ Race::RAND };
```
Comment (ihhub) [inline src/fheroes2/maps/map_format_info.h:138]: Let's assign this value to 0 and do not use race here. Value 0 corresponds to NONE as a race which is even better to have to detect invalid values.

Also, I suggest to name this member as **race**. Simple as it sounds. Then we won't need an extensive comment for this. Moreover, we can also use this member to check that the hero's race is indeed the same as the sprites for them.

File: src/fheroes2/heroes/heroes.cpp [hunk 3]
```diff
@@ -839,6 +845,10 @@ Maps::Map_Format::HeroMetadata Heroes::getHeroMetadata() const
     // Hero's spell points.
     heroMetadata.magicPoints = ( magic_point == UINT32_MAX ) ? static_cast<int16_t>( -1 ) : static_cast<int16_t>( magic_point );
 
+    if ( Modes( JAIL ) ) {
+        heroMetadata.jailedHeroRace = static_cast<uint8_t>( _race );
+    }
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes.cpp:850]: As per proposal below we can simply do this:
```cpp
heroMetadata.race = static_cast<uint8_t>( _race );
```

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I put several suggestions here. Would you mind to take a look at them?

File: src/fheroes2/maps/map_format_info.h (hunk 2 above)
Comment (Districh-ru) [reply src/fheroes2/maps/map_format_info.h]: Hi, @ihhub, sure!
I also had thoughts about using this variable to store the normal hero's race but left it unchanged.

File: src/fheroes2/maps/map_object_info.h (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/maps/map_object_info.h]: OK)
I just moved this function without reviewing and changing it. :(

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]
Hi, @Districh-ru 
Everything works fine!
Just have a tiny question regarding faction name.
image

Not sure it is going to look better, but have you tried to place faction name a little lower? Currently it is placed just above the bottom line of background crown image. For me personally, it looks like the text underlining, which is not pleasant to eye. Could it look better, of the name is shifted ~5px lower?

**Comment by Districh-ru** [comment]
Hi, @Branikolog 
> Could it look better, of the name is shifted ~5px lower?

Yes, it looks a little better. We can replace the background image later if we find a suitable one. :)

изображение

**Comment by ihhub** [comment]
@Districh-ru , many thanks for this update!

**Pull request merged**